src/dataviews/helpers/histogram-helper.js

```diff
@@ -83,47 +72,46 @@ helper.hasChangedSomeOf = function (list, changed) {
   });
 };
 
-helper.calculateLimits = function (bins) {
-  var start = Infinity;
-  var end = -Infinity;
-
-  _.each(bins, function (bin) {
-    if (_.isFinite(bin.min)) {
-      start = Math.min(bin.min, start);
-    }
-    if (_.isFinite(bin.max)) {
-      end = Math.max(bin.max, end);
-    }
-  });
-
-  return {
-    start: start !== Infinity ? start : null,
-    end: end !== -Infinity ? end : null
-  };
-};
-
-helper.calculateDateRanges = function (min, max) {
-  var startDate = moment.unix(min).utc();
-  var endDate = moment.unix(max).utc();
-  var ranges = {};
-
-  _.each(_.keys(MOMENT_AGGREGATIONS), function (agg) {
-    var startClone = startDate.clone();
-    var endClone = endDate.clone();
-    var start = startClone.startOf(agg).unix();
-    var end = endClone.startOf(agg).add(1, MOMENT_AGGREGATIONS[agg]).unix() - 1;
-
-    ranges[agg] = {
-      start: start,
-      end: end
-    };
-  });
+/* Internal functions */
+
+/**
+ * Add a `number` of aggregations to the provided timestamp
+ *
+ * @param {number} timestamp - Starting timestamp
+ * @param {number} number - Number of aggregations to add
+ * @param {object} aggregation
+ * @param {string} aggregation.unit - unit of the aggregation
+ * @param {number} aggregation.factor - number of aggretagion units
+ */
+function add (timestamp, number, aggregation) {
```

Comment:
do we have tests for this funcion?

Reply:
No unit tets, but I have tested it, comparing the output results with the moment.js output for different timestamps and different aggregations.

``` js
function test () {
  var timestamp = 12341324;
  var startDate = moment.unix(timestamp).utc();

  console.log('Start test', new Date(timestamp * 1000));

  var agg = 'quarter';

  for (var i = 0; i < 100000; i++) {
    var momentResult = startDate.clone().add(i, MOMENT_AGGREGATIONS[agg]).unix();
    var localResult = addTimestamp(timestamp, i, MOMENT_AGGREGATIONS_DATA[agg]);

    if (localResult !== momentResult) {
      console.log(new Date(localResult * 1000), new Date(momentResult * 1000));
    }
  }

  console.log('End test');
}

Reply:
I think having unit tests for "complex" functions like this is always a good idea

Reply:
+1 for adding some unit tests

Reply:
There you go: https://github.com/CartoDB/cartodb.js/commit/1d2879d13f7353ab85f7abec9420fb13b3a5e3e8